From 6783ff6a1531723c04610f343f57b9dc30333ac4 Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tduszynski@cavium.com>
Date: Fri, 8 Mar 2019 12:00:58 +0100
Subject: [PATCH 022/386] octeontx2-pf: proceed silently if link event response
 is missing

Do not bother anybody with error message showed whenever mbox sync
fails. Just proceed silently.

Change-Id: I4320a64553c61cd818bae721bc68bdd985f0d73d
Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5381
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 2d8902abab16..02e6d9a82540 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1810,7 +1810,7 @@ static void otx2_vf_link_event_task(struct work_struct *work)
 	struct cgx_link_info_msg *req;
 	struct mbox_msghdr *msghdr;
 	struct otx2_nic *pf;
-	int err, vf_idx;
+	int vf_idx;
 
 	config = container_of(work, struct otx2_vf_config,
 			      link_event_work.work);
@@ -1829,10 +1829,7 @@ static void otx2_vf_link_event_task(struct work_struct *work)
 	req->hdr.sig = OTX2_MBOX_REQ_SIG;
 	memcpy(&req->link_info, &pf->linfo, sizeof(req->link_info));
 
-	err = otx2_sync_mbox_up_msg(&pf->mbox_pfvf[0], vf_idx);
-	if (err)
-		dev_warn(pf->dev, "VF%d failed to acknowledge link event\n",
-			 vf_idx);
+	otx2_sync_mbox_up_msg(&pf->mbox_pfvf[0], vf_idx);
 }
 
 static int otx2_sriov_enable(struct pci_dev *pdev, int numvfs)
-- 
2.17.1

